### ISSUE TITLE
Transform on columnIndex with MultiIndex is broken in 1.4.0

### ISSUE THREAD

**Issue by vitaliknet** [bloomberg/ipydatagrid#562]
It appears after 1.4.0 transform refactoring the following code no longer works.

dg_ind.transform([{'type': 'filter', 'columnIndex': 1, 'operator': 'in', 
                              'value': ['A', 'AA']
                          }])

dg_ind.data.index
MultiIndex([('2024-12-16 00:00:00+00:00',    'A'),
            ('2024-12-16 00:00:00+00:00',   'AA'),
            ...
            ('2024-12-16 00:00:00+00:00',  'ZZZ')],
           names=[('date', ''), ('ticker', '')], length=8965)

Something like below does not work either.

dg_ind.transform([{'type': 'filter', 'column': ('ticker', ''), 'operator': 'in', 
                              'value': ['A', 'AA']
                          }])

**Comment by ollyhensby** [comment]
Yes, I am also getting this issue in the latest release (v1.4.0) when trying to use transforms for multi-index columns.

Example code with multi-index:

```py
import pandas as pd

# Sample data
data = {
    ('Sales', '2023'): [200, 220, 250],
    ('Sales', '2024'): [230, 210, 240],
}

# Create a DataFrame
df = pd.DataFrame(data)

# Set MultiIndex for columns
df.columns = pd.MultiIndex.from_tuples(df.columns, names=('Category', 'Year'))
grid = DataGrid(df)
display(grid)
grid.transform(
    [
        {
            "type": "filter",
            "columnIndex": 1,
            "operator": "in",
            "value": "200"
        }
    ]
)
```

Output:

![Image](https://github.com/user-attachments/assets/123414f8-2da6-4e47-aa35-28ccd73509b4)

^ We see that the table has not been transformed and is still showing all of the sample data.

---

Now if I apply the same transform BUT to a grid that is NOT multi-index:

```py
import pandas as pd

# Sample data
data = {
    '2023': [200, 220, 250],
    '2024': [230, 210, 240],
}

# Create a DataFrame
df = pd.DataFrame(data)
grid = DataGrid(df)
grid.transform(
    [
        {
            "type": "filter",
            "columnIndex": 1,
            "operator": "in",
            "value": "200"
        }
    ]
)
display(grid)
```

Output:

![Image](https://github.com/user-attachments/assets/1f9256c3-d5d8-41eb-bae6-8fa2050c4961)

^ This is now transformed as expected.
